be/src/main/java/yeonba/be/notification/service/NotificationService.java

```diff
@@ -0,0 +1,100 @@
+package yeonba.be.notification.service;
+
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import yeonba.be.exception.GeneralException;
+import yeonba.be.exception.NotificationException;
+import yeonba.be.notification.dto.request.NotificationPageRequest;
+import yeonba.be.notification.dto.response.NotificationPageResponse;
+import yeonba.be.notification.dto.response.NotificationUnreadExistResponse;
+import yeonba.be.notification.entity.Notification;
+import yeonba.be.notification.entity.NotificationPermission;
+import yeonba.be.notification.entity.NotificationType;
+import yeonba.be.notification.event.NotificationSendEvent;
+import yeonba.be.notification.repository.NotificationCommand;
+import yeonba.be.notification.repository.NotificationPermissionQuery;
+import yeonba.be.notification.repository.NotificationQuery;
+import yeonba.be.user.entity.User;
+import yeonba.be.user.repository.user.UserQuery;
+
+@Service
+@RequiredArgsConstructor
+public class NotificationService {
+
+    private final UserQuery userQuery;
+    private final NotificationCommand notificationCommand;
+    private final NotificationQuery notificationQuery;
+    private final NotificationPermissionQuery notificationPermissionQuery;
+
+    @Transactional
+    public NotificationPageResponse getRecentlyReceivedNotificationsBy(
+        long receiverId, NotificationPageRequest request) {
+
+        int pageNumber = request.getPage();
+        int size = 9;
+
+        PageRequest pageRequest = PageRequest.of(pageNumber, size);
+
+        User receiver = userQuery.findById(receiverId);
+        Page<Notification> page =
+            notificationQuery.findRecentlyReceivedNotificationsBy(receiver, pageRequest);
+
+        readNotifications(page.getContent());
+
+        return NotificationPageResponse.from(page);
+    }
+
+    private void readNotifications(List<Notification> notifications) {
+
+        notifications.stream()
+            .filter(notification -> !notification.isRead())
+            .forEach(Notification::read);
```

Comment:
아래 코드와 결과적으로 차이가 없을 것같은데 filter()가 들어가 있는 이유는 무엇인가요?

```suggestion
        notifications.stream()
            .forEach(Notification::read);
```

Reply:
불필요한 `UPDATE`쿼리가 발생하지 않도록 읽지 않은 알림을 필터링하는 로직을 추가하였습니다.